src/main/java/com/mongodb/jdbc/MongoResultSetMetaData.java

```diff
@@ -1,28 +1,46 @@
 package com.mongodb.jdbc;
 
+import java.util.List;
+import java.util.HashMap;
+import java.math.BigDecimal;
 import java.sql.ResultSetMetaData;
 import java.sql.SQLException;
 import java.sql.Types;
+import java.sql.Timestamp;
 import org.bson.BsonType;
 
 public class MongoResultSetMetaData implements ResultSetMetaData {
-    private MongoResultDoc mongoResultDoc;
-
+    private List<Column> columns;
+	private HashMap<String, Integer> columnPositions;
```

Comment:
I moved the column position cache inside this class, since column labels are really a property of result set metadata, not the result set itself